app/models/order.rb

```diff
@@ -162,6 +162,8 @@ def backdate_order_details!(update_by, order_status)
     order_status = order_status.is_a?(OrderStatus) ? order_status : OrderStatus.find(order_status)
 
     order_details.each do |od|
+      next if od.reservation # reservations should always have order_status dictated by their dates
```

Comment:
Since this is the main path everything goes through, is there a way to make all of the logic live within this block to make the path a little more readable? Somehow pull the `complete_past_reservations!` into this loop?

I'm probably missing something here, but something like this. Otherwise, we're kind of doing two loops one to (possibly) backdate reservations and then another to backdate orders based on the status chosen in the dropdown. I'm open to a discussion but I feel like one loop with three (well, technically four) outcomes is easier to grok than two loops where one has two outcomes.

```ruby
if od.reservation 
  od.backdate_to_complete! od.reservation.reserve_end_at if od.reservation.reserve_end_at < Time.current
elsif order_status.root == OrderStatus.complete
  od.backdate_to_complete!(ordered_at)
else
  od.update_order_status!(update_by, order_status, admin: true)
end
```

Reply:
We can't do this refactoring because we want `complete_past_reservations!` to run regardless of whether `acting_as? && order_status_id.present?` is true.